### PR TITLE
Scripts/Spells: Fix Rogue Stealth with cleanup for dropped spells and…

### PR DESCRIPTION
**Pull request by aquadeus**
… deleting related spells to move them in their own scripts

**Changes proposed:**

-  Fix Rogue Stealth with cleanup for dropped spells and deleting related spells to move them in their own scripts

**Tests performed:**

Tested in-game

### CODE REVIEW

**Review by Shauren — CHANGES_REQUESTED** [review]
What is with this PR? I can see Master of Subtlety and Death from Above being removed from the game but Shadow Focus and Premeditation are still very much part of Subtlety talent tree.

Don't just completely remove their code without also adding their own scripts as you wrote in PR description

**Comment by aquadeus** [comment]
mb i had the scripts done but i forgot to push them

**Comment by aquadeus** [comment]
added missing scripts

**Comment by Aokromes** [comment]
Conflicting files
src/server/scripts/Spells/spell_rogue.cpp

**Comment by aquadeus** [comment]
done

**Comment by Aokromes** [comment]
/home/runner/work/TrinityCore/TrinityCore/src/server/scripts/Spells/spell_rogue.cpp:841:2: error: expected ‘;’ after class definition
  841 | }
      |  ^
      |  ;
compilation terminated due to -Wfatal-errors.

File: sql/updates/world/master/2024_99_99_99_shadow_focus_and_premeditation.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DELETE FROM `spell_proc` WHERE `SpellId` IN (343173);
+INSERT INTO `spell_proc` (`SpellId`,`SchoolMask`,`SpellFamilyName`,`SpellFamilyMask0`,`SpellFamilyMask1`,`SpellFamilyMask2`,`SpellFamilyMask3`,`ProcFlags`,`ProcFlags2`,`SpellTypeMask`,`SpellPhaseMask`,`HitMask`,`AttributesMask`,`DisableEffectsMask`,`ProcsPerMinute`,`Chance`,`Cooldown`,`Charges`) VALUES
+(343173,0x00,8,0x00000000,0x00400000,0x00000000,0x00000000,0x0,0x0,0x1,0x2,0x0,0x0,0x0,0,0,0,0); -- Premeditation
```
Comment (Shauren) [inline sql/updates/world/master/2024_99_99_99_shadow_focus_and_premeditation.sql:3]: This mask includes too many spells (finishing moves, instant poison damage proc)

File: sql/updates/world/master/2024_99_99_99_shadow_focus_and_premeditation.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+DELETE FROM `spell_script_names` WHERE `ScriptName` IN ('spell_rog_premeditation', 'spell_rog_shadow_focus');;
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(1784, 'spell_rog_premeditation'),
+(1784, 'spell_rog_shadow_focus'),
+(185313, 'spell_rog_shadow_focus');
```
Comment (Shauren) [inline sql/updates/world/master/2024_99_99_99_shadow_focus_and_premeditation.sql:9]: spell_rog_premeditation_proc not assigned

**Review by Shauren — CHANGES_REQUESTED** [review]

**Pull request merged**